qiskit/quantum_info/operators/random.py

```diff
@@ -36,18 +36,18 @@ def random_unitary(dims, seed=None):
 
     Args:
         dims (int or tuple): the input dimensions of the Operator.
-        seed (int or RandomState): Optional. Set a fixed seed or
+        seed (int or default_rng): Optional. Set a fixed seed or
                                    generator for RNG.
 
     Returns:
         Operator: a unitary operator.
     """
     if seed is None:
-        random_state = np.random
-    elif isinstance(seed, RandomState):
+        random_state = np.random.default_rng()
+    elif isinstance(seed, np.random.Generator):
```

Comment:
The reason for the `None` case before was to bypass the slow initialization of `RandomState`. This shouldn't be needed now with `default_rng` so all these if blocks can be simplified to:

```python
if isinstance(seed, np.random.Generator):
    random_state = seed
else:
    random_state = np.random.default_rng(seed)
```

Reply:
If we wanted to speed up these functions even more we could also leave the `None` case and assign it to a global variable `default_rng` initialied on import of the random.py module.

ie.
```python
DEFAULT_RNG = default_rng()

def  random_unitary(dims, seed=None):
    if seed is None:
        random_state = DEFAULT_RNG
    elif...
```

Reply:
Your more than welcome to in another PR.  This one was fun enough.

Reply:
Out of interest I timed these two cases (initializing new default each function call, vs using a module level default rng) and got this:

![rng_init](https://user-images.githubusercontent.com/2235104/80511639-5d0deb80-894a-11ea-8da4-036a9adfa2c8.png)

Reply:
this is nice but its a new pr